internal/website/content/howto/docstore/_index.md

```diff
@@ -1,14 +1,359 @@
+[`mongodocstore.URLOpener`][].
+
+[`mongodocstore.URLOpener`]:  https://godoc.org/gocloud.dev/docstore/mongodocstore#URLOpener
+
+#### MongoDB Constructors {#mongo-ctor}
+
+The [`mongodocstore.OpenCollection`][] constructor opens a MongoDB collection.
+You must first obtain a standard MongoDB Go client using
+[`mongodocstore.Dial`][] or the package
+[`go.mongodb.org/mongo-driver/mongo`](https://godoc.org/go.mongodb.org/mongo-driver/mongo).
+Obtain a `*mongo.Collection` from the client with
+`client.Database(dbName).Collection(collName)`. Then pass the result to
+`mongodocstore.OpenCollection` along with the name of the ID field, or `""` to
+use `_id`.
+
+{{< goexample "gocloud.dev/docstore/mongodocstore.ExampleOpenCollection" >}}
+
+Instead of mapping the document ID to a field, you can supply a function to
+construct the ID from the document contents with
+[`mongodocstore.OpenCollectionWithIDFunc`][]. This can be useful for documents
+whose name is the combination of two or more fields.
+
+{{< goexample
+"gocloud.dev/docstore/mongodocstore.ExampleOpenCollectionWithIDFunc" >}}
+
+[`mongodocstore.Dial`]: https://godoc.org/gocloud.dev/docstore/mongodocstore#Dial
+[`mongodocstore.OpenCollection`]: https://godoc.org/gocloud.dev/docstore/mongodocstore#OpenCollection
+[`mongodocstore.OpenCollectionWithIDFunc`]: https://godoc.org/gocloud.dev/docstore/mongodocstore#OpenCollectionWithIDFunc
+
+### In-Memory Document Store {#mem}
+
+The [`memdocstore`](https://godoc.org/gocloud.dev/docstore/mongodocstore)
+package implements an in-memory document store suitable for testing and
+development.
+
+URLs for the in-memory store have a `mem:` scheme and provide the name of the
+document field to use as a primary key. There is no collection name; each call
+to one of the `OpenCollection` functions creates a distinct collection.
+
+{{< goexample
+"gocloud.dev/docstore/memdocstore.Example_openCollectionFromURL" >}}
+
+Full details about acceptable URLs can be found under the API reference for
+[`memdocstore.URLOpener`][].
+
+[`memdocstore.URLOpener`]:  https://godoc.org/gocloud.dev/docstore/memdocstore#URLOpener
+
+#### Mem Constructors {#mem-ctor}
+
+The [`memdocstore.OpenCollection`][] constructor creates and opens a collection,
+taking the name of the key field.
+
+{{< goexample "gocloud.dev/docstore/memdocstore.ExampleOpenCollection" >}}
+
+You can instead supply a function to construct the primary key from the document
+contents with [`memdocstore.OpenCollectionWithKeyFunc`][]. This can be useful
+for documents whose name is the combination of two or more fields.
+
+{{< goexample
+"gocloud.dev/docstore/memdocstore.ExampleOpenCollectionWithKeyFunc" >}}
+
+[`memdocstore.OpenCollection`]: https://godoc.org/gocloud.dev/docstore/memdocstore#OpenCollection
+[`memdocstore.OpenCollectionWithKeyFunc`]: https://godoc.org/gocloud.dev/docstore/memdocstore#OpenCollectionWithKeyFunc
+
+## Actions {#actions}
```

Comment:
There is supposed to be a section "Using a Collection" above the services that hosts all this other content. See blob page structure.